Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -357,6 +357,21 @@ static PlatformTimeRanges removeSamplesFromTrackBuffer(const DecodeOrderSampleMa
     return trackBuffer.removeSamples(samples, logPrefix);
 }
 
+MediaTime SourceBufferPrivate::findPreviousSyncSamplePresentationTime(const MediaTime& time)
+{
+    MediaTime previousSyncSamplePresentationTime = time;
+    for (auto& trackBufferKeyValue : m_trackBufferMap) {
+        TrackBuffer& trackBuffer = trackBufferKeyValue.value;
+        auto sampleIterator = trackBuffer.samples().decodeOrder().findSyncSamplePriorToPresentationTime(time);
+        if (sampleIterator == trackBuffer.samples().decodeOrder().rend())
+            continue;
+        const MediaTime& samplePresentationTime = sampleIterator->first.second;
+        if (time > samplePresentationTime)
```

Comment:
Nit (feel free to ignore): This is a classic "find minimum" algorithm. Flipping the condition to `samplePresentationTime < time` feels more natural to me, whereas the other way in first read makes me think it's an unrelated condition. Maybe it's just me though.